Jenkinsfile_parameterized

```diff
@@ -0,0 +1,18 @@
+#!groovy
+@Library(value="Infrastructure", changelog=false) _
+
+properties([
+  parameters([
+    string(name: 'PRODUCT_NAME', defaultValue: 'rpe', description: ''),
+    string(name: 'APP', defaultValue: 'bulk-scanning', description: ''),
+    string(name: 'TYPE', defaultValue: 'java', description: ''),
+    string(name: 'ENVIRONMENT', defaultValue: 'sandbox', description: 'Environment where code should be built and deployed'),
+    choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
+  ]),
+  [$class: 'GithubProjectProperty', projectUrlStr: 'github.com/hmcts/bulk-scanning'],
+  pipelineTriggers([[$class: 'GitHubPushTrigger']])
+])
+
+withParameterizedPipeline(params.TYPE, params.PRODUCT_NAME, params.APP, params.ENVIRONMENT, params.SUBSCRIPTION) {
+
```

Comment:
You could add  ` enableSlackNotifications(channel)` so that we know when build fails